src/cmds/fs/cat.c

```diff
@@ -21,13 +21,20 @@ int main(int argc, char **argv) {
 	int number = 0, line = 0, new_line = 1;
 	FILE *fd;
 	char buff;
+	int no_of_arg;
+	char *file_names[argc];
+	size_t arg_len;
 
+	for (no_of_arg = 0; no_of_arg < argc; no_of_arg++) {
+		file_names[no_of_arg] = strdup (argv[no_of_arg]);
```

Comment:
Do we really need to create a copy for those strings? As I can see, they are just passed to `strcmp()` and `fopen()`, both of them have `const` modifier for arguments, so they should not change this memory anyway.
```
int strcmp(const char *s1, const char *s2);
FILE *fopen(const char *pathname, const char *mode);
```